pkg/console/operator/sync_v400.go

```diff
@@ -49,46 +50,72 @@ func sync_v400(co *consoleOperator, operatorConfig *operatorv1.Console, consoleC
 
 	rt, rtChanged, rtErr := SyncRoute(co, operatorConfig)
 	if rtErr != nil {
+		co.operatorStatusResourceSyncFailure(operatorConfig, fmt.Sprintf("%v: %s\n", "route", rtErr))
 		return operatorConfig, consoleConfig, toUpdate, rtErr
 	}
 	toUpdate = toUpdate || rtChanged
 
 	_, svcChanged, svcErr := SyncService(co, recorder, operatorConfig)
 	if svcErr != nil {
+		co.operatorStatusResourceSyncFailure(operatorConfig, fmt.Sprintf("%q: %v\n", "service", svcErr))
 		return operatorConfig, consoleConfig, toUpdate, svcErr
 	}
 	toUpdate = toUpdate || svcChanged
 
 	cm, cmChanged, cmErr := SyncConfigMap(co, recorder, operatorConfig, consoleConfig, rt)
 	if cmErr != nil {
+		co.operatorStatusResourceSyncFailure(operatorConfig, fmt.Sprintf("%q: %v\n", "configmap", cmErr))
 		return operatorConfig, consoleConfig, toUpdate, cmErr
 	}
 	toUpdate = toUpdate || cmChanged
 
 	serviceCAConfigMap, serviceCAConfigMapChanged, serviceCAConfigMapErr := SyncServiceCAConfigMap(co, operatorConfig)
 	if serviceCAConfigMapErr != nil {
+		co.operatorStatusResourceSyncFailure(operatorConfig, fmt.Sprintf("%q: %v\n", "serviceCAconfigmap", serviceCAConfigMapErr))
 		return operatorConfig, consoleConfig, toUpdate, serviceCAConfigMapErr
 	}
 	toUpdate = toUpdate || serviceCAConfigMapChanged
 
 	sec, secChanged, secErr := SyncSecret(co, recorder, operatorConfig)
 	if secErr != nil {
+		co.operatorStatusResourceSyncFailure(operatorConfig, fmt.Sprintf("%q: %v\n", "secret", secErr))
 		return operatorConfig, consoleConfig, toUpdate, secErr
 	}
 	toUpdate = toUpdate || secChanged
 
 	_, oauthChanged, oauthErr := SyncOAuthClient(co, operatorConfig, sec, rt)
 	if oauthErr != nil {
+		co.operatorStatusResourceSyncFailure(operatorConfig, fmt.Sprintf("%q: %v\n", "oauth", oauthErr))
 		return operatorConfig, consoleConfig, toUpdate, oauthErr
 	}
 	toUpdate = toUpdate || oauthChanged
 
-	_, depChanged, depErr := SyncDeployment(co, recorder, operatorConfig, cm, serviceCAConfigMap, sec)
+	actualDeployment, depChanged, depErr := SyncDeployment(co, recorder, operatorConfig, cm, serviceCAConfigMap, sec)
 	if depErr != nil {
+		co.operatorStatusResourceSyncFailure(operatorConfig, fmt.Sprintf("%q: %v\n", "route", depErr))
 		return operatorConfig, consoleConfig, toUpdate, depErr
 	}
+
 	toUpdate = toUpdate || depChanged
 
+	// at this point, we should not be failing anymore
+	co.operatorStatusResourceSyncSuccess(operatorConfig)
+
+	// but we may be in a transitional state, if any of the above resources changed
+	if toUpdate {
+		co.operatorStatusProgressing(operatorConfig)
```

Comment:
was debugging a bit more and figured our that even if none of the resources are updated, here the `toUpdate` will be set to `true`, which shouldn't happen if nothing gets updated, right ?

Reply:
Nope, if nothing is updated, we want it `false`.  So that is definitely something worth tracking down.